### PR TITLE
Complete intro/1-15

### PR DESCRIPTION
**Pull request by haroldrandom**
Complete intro/[0..15].

### CODE REVIEW

File: intro/04_data_structures/queen.rb [hunk 1]
```diff
@@ -6,4 +6,16 @@
          "No escape from reality."
 
 # Your code goes here
+marks = Hash.new(0)
 
+(0...lyrics.length ).each { |i|
```
Comment (matthewmpalen) [inline intro/04_data_structures/queen.rb:11]: Looks good. You could also do

```ruby
lyrics.split("").each do |i|
```

where `i` is an individual character

File: intro/07_symbols/sites.rb [hunk 2]
```diff
@@ -7,4 +7,5 @@
 ]
 
 # Your code goes here
-
+url_map = Hash.new
+urls.each { |url| url_map[url.to_sym] = open(url) }
```
Comment (matthewmpalen) [inline intro/07_symbols/sites.rb]: Close, but slightly different. `open(url)` will give you the response of the webpage but not its size. You need to add a little more here.

File: intro/10_scope/basic_game.rb [hunk 3]
```diff
@@ -1,9 +1,35 @@
...
+    def initialize(name)
+        @name = name
+        @players = Array.new
+    end
+
+    def Game.MAX_PLAYER_COUNT
+        @@MAX_PLAYER_COUNT
+    end
+
+    def add_player(player)
```
Comment (matthewmpalen) [inline intro/10_scope/basic_game.rb:23]: It seems like you might have slightly misunderstood the purpose of `MAX_PLAYER_COUNT` (and I might not have explained it well enough). When adding a player to the game, you should first check to see if there are already the max number of players. If so, do not add the next player (perhaps log an error or tell the user why it didn't work).

**Review by matthewmpalen — COMMENTED** [review]
Looks mostly good. I noticed you didn't try lession 5 though. Had problems?

File: intro/10_scope/basic_game.rb (hunk 3 above)
Comment (haroldrandom) [inline intro/10_scope/basic_game.rb]: Got it. I will fix it in another pull requests.

Comment (haroldrandom) [reply intro/10_scope/basic_game.rb:23]: Got it. I will fix it in another commit.

File: intro/07_symbols/sites.rb (hunk 2 above)
Comment (haroldrandom) [reply intro/07_symbols/sites.rb]: Oh, a mistake, will be fixed in next commit.

**Comment by haroldrandom** [comment]
Finish 01 - 15 tasks.

Besides, in section intro/16_yaml order.rb, after seaching `truncate(2)` and to_s("F") in Ruby docs, I still can't find any documents related with those two functions in that context.

Could you give me some hints?

**Pull request closed without merging**